config/jobs/kubernetes/kops/build_grid.py

```diff
@@ -141,37 +143,68 @@ def should_skip_newer_k8s(k8s_version, kops_version):
         return True
     return float(k8s_version) > float(kops_version)
 
+def latest_aws_image(owner, name):
+    client = boto3.client('ec2', region_name='us-east-1')
+    response = client.describe_images(
+        Owners=[owner],
+        Filters=[
+            {
+                'Name': 'name',
+                'Values': [
+                    name,
+                ],
+            },
+        ],
+    )
+    images = {}
+    for image in response['Images']:
+        images[image['CreationDate']] = image['ImageLocation']
+    return images[sorted(images, reverse=True)[0]]
+
+distro_images = {
+    'amzn2': latest_aws_image('137112412989', 'amzn2-ami-hvm-*-x86_64-gp2'),
+    'centos7': latest_aws_image('125523088429', 'CentOS 7.*x86_64'),
+    'centos8': latest_aws_image('125523088429', 'CentOS 8.*x86_64'),
+    'deb9': latest_aws_image('379101102735', 'debian-stretch-hvm-x86_64-gp2-*'),
+    'deb10': latest_aws_image('136693071363', 'debian-10-amd64-*'),
+    'flatcar': latest_aws_image('075585003325', 'Flatcar-stable-*-hvm'),
+    'rhel7': latest_aws_image('309956199498', 'RHEL-7.*_HVM_*-x86_64-0-Hourly2-GP2'),
+    'rhel8': latest_aws_image('309956199498', 'RHEL-8.*_HVM-*-x86_64-0-Hourly2-GP2'),
+    'u1804': latest_aws_image('099720109477', 'ubuntu/images/hvm-ssd/ubuntu-bionic-18.04-amd64-server-*'), # pylint: disable=line-too-long
+    'u2004': latest_aws_image('099720109477', 'ubuntu/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-*'), # pylint: disable=line-too-long
+}
+
 def distro_info(distro):
     if distro == 'amzn2':
         kops_ssh_user = 'ec2-user'
-        kops_image = '137112412989/amzn2-ami-hvm-2.0.20201126.0-x86_64-gp2'
+        kops_image = distro_images[distro]
```

Comment:
can we just consolidate all of these to a single `kops_image = distro_images[distro]` at the top of this function?

Reply:
in fact we could add the ssh user mapping to distro_images and then get rid of this function entirely

Reply:
I think this last commit should do it :).